src/features/views/store.ts

```diff
@@ -180,6 +178,46 @@ const viewsSlice = createSlice({
         }
       }
     },
+    columnOrderUpdated: (state, action: PayloadAction<[number, number[]]>) => {
+      const [viewId, columnOrder] = action.payload;
+      // Re-arrange columns
+      const colList = state.columnsByViewId[viewId];
+      if (colList) {
+        const newColListItems = columnOrder.map((colId) => {
+          const col = colList.items.find((col) => col.id == colId);
+          if (col) {
+            return col;
+          } else {
+            throw new Error('Could not find column!');
+          }
+        });
+
+        // Re-arrange columns of data-rows
+        const rowList = state.rowsByViewId[viewId];
+        if (rowList) {
+          const newRowListItems = rowList.items.map((row) => {
+            if (row.data) {
+              return {
+                ...row,
+                data: {
+                  content: columnOrder.map((colId) => {
+                    const idx = colList.items.findIndex(
+                      (col) => col.id == colId
+                    )!;
+                    return row.data?.content[idx];
+                  }),
+                  id: row.data.id,
+                },
+              };
+            } else {
+              return row;
+            }
+          });
+          state.columnsByViewId[viewId].items = newColListItems;
+          state.rowsByViewId[viewId].items = newRowListItems;
+        }
+      }
```

Comment:
Beautiful! This is a big improvement over Gen2 where the data had to be reloaded after reordering columns. Nice work!